cql3/expr/expression.hh

```diff
@@ -302,6 +302,10 @@ column_value_tuple::column_value_tuple(Range r)
         : column_value_tuple(std::vector<column_value>(r.begin(), r.end())) {
 }
 
+// Extracts all binary operators which have the given column on their left hand side.
+// Includes both single-column and multi-column restrictions.
+// Does not include token() restrictions.
+extern std::vector<expression> extract_column_restrictions(const expression&, const column_definition&);
```

Comment:
Extraneous `extern`

Reply:
Should we return a vector here, or perhaps a single expression that can be a conjunction?

Reply:
I added the `extern` because other functions in the header had it, but it looks like it is not needed.